### PR TITLE
[Issue #1138] Added 'Rename' option to right-click menu for Documents tab

### PR DESCRIPTION
**Pull request by Fordi**

### CODE REVIEW

File: src/document.c [hunk 1]
```diff
@@ -1273,7 +1273,6 @@ void document_show_tab(GeanyDocument *doc)
 		document_get_notebook_page(doc));
 }
 
-
```
Comment (b4n) [inline src/document.c]: don't remove this line (it's part of the style -- and in any case, don't introduce style-only changes in the middle of a commit)

File: src/document.c [hunk 2]
```diff
@@ -1656,6 +1655,12 @@ gboolean document_reload_prompt(GeanyDocument *doc, const gchar *forced_enc)
 	g_free(base_name);
 	return result;
 }
+/* also used for reloading when forced_enc is NULL */
+gboolean document_rename_prompt(GeanyDocument *doc) {
```
Comment (b4n) [inline src/document.c]: I don't think this convenience function should be added.  Not only it's very simple and only use "public" API, but it's also only used in one place.

File: src/document.c [hunk 3]
```diff
@@ -1656,6 +1655,12 @@ gboolean document_reload_prompt(GeanyDocument *doc, const gchar *forced_enc)
...
+	document_show_tab(doc);
+	dialogs_show_save_as();
+	return FALSE;
```
Comment (b4n) [inline src/document.c]: why return something if it's always `FALSE`?  if anything, it should be `return dialogs_show_save_as();`

File: src/document.c (hunk 2 above)
Comment (b4n) [inline src/document.c]: Also the name is partially misleading, it's the same for save as/rename

File: src/sidebar.c [hunk 4]
```diff
@@ -724,6 +725,16 @@ static void create_openfiles_popup_menu(void)
 			G_CALLBACK(on_openfiles_document_action), GINT_TO_POINTER(OPENFILES_ACTION_RELOAD));
 	doc_items.reload = item;
 
+	item = gtk_image_menu_item_new_with_mnemonic(_("R_ename or Save As..."));
```
Comment (b4n) [inline src/sidebar.c]: We use *Save As* alone in the rest of the UI, should we introduce *Rename* here?

File: src/sidebar.c [hunk 5]
```diff
@@ -724,6 +725,16 @@ static void create_openfiles_popup_menu(void)
...
+	gtk_image_menu_item_set_image(GTK_IMAGE_MENU_ITEM(item),
+		gtk_image_new_from_stock(GTK_STOCK_EDIT, GTK_ICON_SIZE_MENU));
```
Comment (b4n) [inline src/sidebar.c]: …and it affects the icon too

**Comment by b4n** [comment]
Although I made a lot of comments, it looks mostly fine.

**Comment by Fordi** [comment]
This was just a spike with reference to the ticket (#1138) to provide a quick implementation; I didn't vet it fully; the impl was to add it for my local machine, and I floated it just to see if you guys would want it for 1138.

Initially the idea was to shove an in-place `Rename` text box in the Documents list when you click rename - that's why the edit is a bit more complex / inconsistent than it needs to be - but when I noticed you could do it from the `Save As...` dialog, I went the lazy route (I had non-voluntary work to get finished) instead.

Anyway, I'll close the PR if y'all don't want it.

**Comment by b4n** [comment]
No, it makes sense to have it next to *Save* if people want it, so after a few fixes this PR would be fine.

**Comment by kugel-** [comment]
It doesn't look like the requested changes will be made. I suggest closing this.

**Pull request closed without merging**